Chessington.GameEngine/Pieces/Pawn.cs

```diff
@@ -14,11 +14,66 @@ public override IEnumerable<Square> GetAvailableMoves(Board board)
             
             if (this.Player == Player.White)
```

Comment:
This is a very long method that's difficult to follow. Consider splitting it into 2 or more private methods